web/Controllers/LandlordController.cs

```diff
@@ -51,29 +57,47 @@ public async Task<ActionResult> RegisterPost([FromForm] LandlordProfileModel cre
         }
 
         var user = GetCurrentUser();
+        ILandlordService.LandlordRegistrationResult result;
+        LandlordDbModel? landlord = null;
 
-        var result = await _landlordService.RegisterLandlordWithUser(createModel, user);
+        if (createModel.Unassigned && user.IsAdmin)
+        {
+            (result, landlord) = await _landlordService.RegisterLandlord(createModel);
+        }
+        else
+        {
+            return StatusCode(403);
+        }
 
         switch (result)
         {
             case ILandlordService.LandlordRegistrationResult.Success:
                 _logger.LogInformation("Successfully created landlord for user {UserId}", user.Id);
-                string msgBody = $"A Landlord has just registered\n"
-                                 + "\n"
-                                 + $"Title: {createModel.Title}\n"
-                                 + $"First Name: {createModel.FirstName}" + "\n"
-                                 + $"Last Name: {createModel.LastName}" + "\n"
-                                 + $"Company Name: {createModel.CompanyName}" + "\n"
-                                 + $"Email: {createModel.Email}" + "\n"
-                                 + $"Phone: {createModel.Phone}" + "\n";
+                var msgBody = "A Landlord has just registered\n"
+                              + "\n"
+                              + $"Title: {createModel.Title}\n"
+                              + $"First Name: {createModel.FirstName}" + "\n"
+                              + $"Last Name: {createModel.LastName}" + "\n"
+                              + $"Company Name: {createModel.CompanyName}" + "\n"
+                              + $"Email: {createModel.Email}" + "\n"
+                              + $"Phone: {createModel.Phone}" + "\n";
                 _mailService.SendMsg(msgBody);
+                if (!createModel.Unassigned)
+                {
+                    return Redirect(Url.Action("MyProfile")!);
+                }
 
-                return Redirect(Url.Action("MyProfile")!);
+                if (landlord == null)
+                {
+                    return StatusCode(500);
+                }

```

Comment:
I'd rather just let it throw an exception on the next line, that's easier to debug than a naked 500 response